pyspecdata/plot_funcs/image.py

```diff
@@ -150,15 +150,21 @@ def axis_labeler(x): # below, I turn off math mode for the axis names
         ax.set_xlim((max(these_xlims),min(these_xlims)))
     return retval
 
-def imagehsv(A,logscale = False,black = False):
+def imagehsv(A, logscale=False, black=False, scaling=None):
     "This provides the HSV mapping used to plot complex number"
     # compare to http://www.rapidtables.com/convert/color/hsv-to-rgb.htm
+    A = A.copy()
     n = 256
     mask = np.isnan(A)
     A[mask] = 0
+    if scaling is None:
+        A /= abs(A).max()
+    else:
+        A /= scaling
+        mask |= abs(A) > 1.0
+        A[mask] = 0
     mask = mask.reshape(-1,1)
     intensity = abs(A).reshape(-1,1)
-    intensity /= abs(A).max()
```

Comment:
this upgrade is important to implementing DCCT, no?

Reply:
yes this is the whole scaling issue. we can't scale without these changes